codegen/src/codegen_rust.rs

```diff
@@ -1212,7 +1212,7 @@ impl<'model> RustCodeGen<'model> {
         writeln!(
             w,
             r#"/// client for sending {} messages
-              #[derive(Debug)]
+              #[derive(Clone, Debug)]
```

Comment:
Isn't this going to cause build failures? `T` isn't clone, so it can't derive

Reply:
fixed by merging feat/smithy-bindgen branch where I had already applied this fix

Reply:
I'm still seeing `T` below only have a `Transport` bound rather than `Transport + Clone`

Reply:
Oh wait, I found where it is at. Transport must implement Clone